lib/Guzzle/AuthenticatorSubscriber.php

```diff
@@ -83,6 +83,8 @@ public function loginIfRequested(CompleteEvent $event)
             $emitter->attach($this);
 
             $event->retry();
+
+            ++self::$retries;
```

Comment:
I'd replace the condition above with: 

´´´php
if (++self::$retries >= self::MAX_TRIES) {
  break;
}
´´´